package valhalla.odin;

message DirectionsOptions {
  
  enum Units {
    kKilometers = 0;
    kMiles = 1;
  }
  
  enum Format {
    json = 0;
    gpx = 1;
    osrm = 2;
  }

  optional Units units = 1;                         // kKilometers or kMiles
  optional string language = 2 [default = "en-US"]; // Based on IETF BCP 47 language tag string
  optional bool narrative = 3 [default = true];     // Enable/disable narrative production
  optional Format format = 4 [default = json];      // What the response format should be
  optional string id = 5;                           // Optional id for the request
  optional string jsonp = 6;                        // Optional javascript callback for the request
}
